pkg/controllermanager/controller/secretbinding/reconciler_test.go

```diff
@@ -139,85 +116,62 @@ var _ = Describe("SecretBindingControl", func() {
-
-			c.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.AssignableToTypeOf(&gardencorev1beta1.SecretBinding{})).DoAndReturn(func(_ context.Context, _ client.ObjectKey, obj *gardencorev1beta1.SecretBinding, _ ...client.GetOption) error {
-				secretBinding.DeepCopyInto(obj)
-				return nil
-			})
-
-			c.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.AssignableToTypeOf(&corev1.Secret{})).DoAndReturn(func(_ context.Context, _ client.ObjectKey, obj *corev1.Secret, _ ...client.GetOption) error {
-				secret.DeepCopyInto(obj)
-				return nil
-			})
-
-			c.EXPECT().Patch(gomock.Any(), gomock.AssignableToTypeOf(&gardencorev1beta1.SecretBinding{}), gomock.Any()).DoAndReturn(
-				func(_ context.Context, sb *gardencorev1beta1.SecretBinding, _ client.Patch, _ ...client.PatchOption) error {
-					*secretBinding = *sb
-					return nil
-				},
-			).AnyTimes()
-
-			c.EXPECT().Patch(gomock.Any(), gomock.AssignableToTypeOf(&corev1.Secret{}), gomock.Any()).DoAndReturn(
-				func(_ context.Context, s *corev1.Secret, _ client.Patch, _ ...client.PatchOption) error {
-					*secret = *s
-					return nil
-				},
-			).AnyTimes()
 		})
 
 		It("should add the label to the secret referred by the secretbinding", func() {
 			_, err := reconciler.Reconcile(ctx, request)
 			Expect(err).NotTo(HaveOccurred())
 
-			expectedLabels := map[string]string{
-				"reference.gardener.cloud/secretbinding": "true",
-			}
-
-			Expect(secret.ObjectMeta.Labels).To(Equal(expectedLabels))
+			Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)).To(Succeed())
+			Expect(secret.ObjectMeta.Labels).To(HaveKeyWithValue(
+				"reference.gardener.cloud/secretbinding", "true",
+			))
 		})
 
 		It("should remove the label from the secret when there are no secretbindings referring it", func() {
-			secretBinding.DeletionTimestamp = &metav1.Time{Time: time.Date(1, 1, 1, 1, 1, 1, 1, time.UTC)}
-
-			c.EXPECT().List(gomock.Any(), gomock.AssignableToTypeOf(&gardencorev1beta1.SecretBindingList{})).DoAndReturn(func(_ context.Context, list *gardencorev1beta1.SecretBindingList, _ ...client.ListOption) error {
-				(&gardencorev1beta1.SecretBindingList{Items: []gardencorev1beta1.SecretBinding{*secretBinding}}).DeepCopyInto(list)
-				return nil
-			}).AnyTimes()
+			_, err := reconciler.Reconcile(ctx, request)
+			Expect(err).NotTo(HaveOccurred())
 
-			c.EXPECT().List(gomock.Any(), gomock.AssignableToTypeOf(&gardencorev1beta1.ShootList{})).DoAndReturn(func(_ context.Context, list *gardencorev1beta1.ShootList, _ ...client.ListOption) error {
-				(&gardencorev1beta1.ShootList{}).DeepCopyInto(list)
-				return nil
-			})
+			Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(secretBinding), secretBinding)).To(Succeed())
+			secretBinding.DeletionTimestamp = &metav1.Time{Time: time.Date(1, 1, 1, 1, 1, 1, 1, time.UTC)}
+			// Add dummy finalizer to prevent deletion
+			secretBinding.Finalizers = append(secretBinding.Finalizers, "finalizer")
```

Comment:
I don't understand why this finalizer is required. This SecretBinding isn't checked afterwards.

Reply:
looks looks good thanks 😄